core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -781,6 +796,318 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
+            if (topmost) { // need to deal with blocking and closing connection
+                if (blocking) {
+                    val timeout = maxWaitForBlockingActivation + blockingInvokeGrace
+                    val futureSeqResultTimeout = futureSeqResult withTimeout(timeout, new BlockingInvokeTimeout(seqActivationId))
+                    // if the future fails with a timeout, the failure is dealt with at the caller level
+                    futureSeqResultTimeout map { wskActivationTuples =>
+                        val wskActivationEithers = wskActivationTuples.map(_._1)
+                        // the execution of the sequence was successful, return the result
+                        val end = Instant.now(Clock.systemUTC())
+                        seqActivation = Some(makeSequenceActivation(user, action, seqActivationId, wskActivationEithers, topmost, cause, start, end))
+                        val atomicActionCnt = wskActivationTuples.last._2
+                        (seqActivationId, seqActivation, atomicActionCnt)
+                    }
+                } else {
+                    // non-blocking sequence execution, return activation id
+                    Future.successful((seqActivationId, None, 0))
+                }
+            } else {
+                // not topmost, no need to worry about terminating incoming request
+                futureSeqResult map { wskActivationTuples =>
+                     val wskActivationEithers = wskActivationTuples.map(_._1)
+                    // all activations are successful, the result of the sequence is the result of the last activation
+                    val end = Instant.now(Clock.systemUTC())
+                    seqActivation = Some(makeSequenceActivation(user, action, seqActivationId, wskActivationEithers, topmost, cause, start, end))
+                    val atomicActionCnt = wskActivationTuples.last._2
+                    (seqActivationId, seqActivation, atomicActionCnt)
+                }
+            }
+
+        // store result of sequence execution
+        // if seqActivation is defined, use it; otherwise create it (e.g., for non-blocking activations)
+        // the execution can reach here without a seqActivation due to non-blocking activations OR blocking activations that reach the blocking invoke timeout
+        // futureSeqResult should always be successful, if failed, there is an error
+        futureSeqResult onComplete {
+            case Success(wskActivationTuples) =>
+                // all activations were successful
+                val activation = seqActivation getOrElse {
+                    val wskActivationEithers = wskActivationTuples.map(_._1)
+                    val end = Instant.now(Clock.systemUTC())
+                    // the response of the sequence is the response of the very last activation
+                    makeSequenceActivation(user, action, seqActivationId, wskActivationEithers, topmost, cause, start, end)
```

Comment:
this can throw an exception in which case no activation will be created - either this should be wrapped with a future itself or a try to allow for recovery.